server/jobs/job_manager.go

```diff
@@ -0,0 +1,77 @@
+package jobs
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/wiggin77/merror"
+
+	"github.com/mattermost/mattermost-plugin-starter-template/server/config"
+)
+
+type Job interface {
+	GetID() string
+	OnConfigurationChange(cfg *config.Configuration) error
+	Stop(timeout time.Duration) error
+}
+
+type JobManager struct {
```

Comment:
Is this basically a goroutine manager for background work? I started reviewing thinking this was based on https://github.com/mattermost/mattermost/blob/master/server/public/pluginapi/cluster/job.go, but then I actually don't know when I would use this. Can you elaborate on the use case for a non-HA aware job manager?